onedal/utils/validation.py

```diff
@@ -438,3 +443,31 @@ def _is_csr(x):
     return isinstance(x, sp.csr_matrix) or (
         hasattr(sp, "csr_array") and isinstance(x, sp.csr_array)
     )
+
+
+def _assert_all_finite(X, allow_nan=False, input_name=""):
+    policy = _get_policy(None, X)
+    params = {
+        "fptype": "float" if X.dtype.name == "float32" else "double",
```

Comment:
What if `_convert_to_supported`  will convert it into float32 on the line 455? Does it make sense to move this params definition after _convert_to_support call?

Reply:
Changed. Thanks for pointing that out.